test/web3_util.py

```diff
@@ -1,34 +1,36 @@
 """Util functions for invoking and calling Web3 contracts"""
 
-from test.test_endpoints import load_file_content
-
-import json
+from solcx import compile_source
 from web3 import Web3
 
 
-def web3_transact(function, url, contract_address, abi_path, *inputs):
-    """Invokes a function in a Web3 contract"""
+def web3_compile(source, path):
+    """Compiles a Solidity contract"""
+    compiled_sol = compile_source(source,output_values=['abi', 'bin'],base_path=path)
+    contract_interface = list(compiled_sol.values())[0]
+    return contract_interface
 
-    web3 = Web3(Web3.HTTPProvider(url))
-    address=Web3.toChecksumAddress(contract_address)
+def web3_deploy(web3: Web3, contract_interface, *inputs):
+    """Deploys a Solidity contract"""
+    abi=contract_interface['abi']
+    bytecode=contract_interface['bin']
+    contract = web3.eth.contract(abi=abi, bytecode=bytecode)
+    tx_hash = contract.constructor(*inputs).transact()
+    tx_receipt = web3.eth.waitForTransactionReceipt(tx_hash)
+    return web3.eth.contract(address=tx_receipt.contractAddress, abi=abi)
 
-    abi=json.loads(load_file_content(abi_path))["abi"]
-    contract = web3.eth.contract(address=address,abi=abi)
+def web3_transact(web3: Web3, function, contract,  *inputs):
+    """Invokes a function in a Web3 contract"""
 
     contract_function = contract.get_function_by_name(function)(*inputs)
-    tx_hash = contract_function.transact({"from": web3.eth.accounts[0], "value": 0})
+    tx_hash = contract_function.transact()
+    web3.eth.waitForTransactionReceipt(tx_hash)
```

Comment:
What happened here, did you find a simple way of sending a transaction?

Reply:
Yes, `WEB3.eth.default_account = WEB3.eth.accounts[0]`, by setting this in the `test_postman.py` file